### ISSUE TITLE
Better output format for RLogger

### ISSUE THREAD

**Issue by eguiraud** [root-project/root#7008]
### Explain what you would like to see improved

Currently, when using RLogger, you get an output such as:

```
Info in <[ROOT.RDF] Info ../../../root_verbose_rdf/tree/dataframe/src/RLoopManager.cxx:658 in void ROOT::Detail::RDF::RLoopManager::Run()>: Starting event loop number 0.
```

Note in particular the redundant `Info`.

### Optional: share how it could be improved

- `s/Info in <(.*)>/\1/`
- not 100% sure about this, but maybe we could shorten the filename to just the part under `$ROOTSYS`

**Comment by Axel-Naumann** [comment]
Thanks! I think I'd like to keep the full path - which can catch cases of "oh that's not the ROOT version I intended". And indeed I should switch to not using `Info` etc, but whatever gets called by `Info`, to provide our own formatting.

**Comment by eguiraud** [comment]
Yes you can catch the odd "wrong ROOT" case, but at what price? :sweat_smile: You get that same (typically super long) path to ROOTSYS printed at every log line, and it makes the actual log much less readable.

Maybe filename/line number could be moved out of the way, to the end of the log line?

**Comment by Axel-Naumann** [comment]
Ideally I'd like to follow the GCC diag format to some extent:
```
<source>:1:1: error: DIAG
```
and that suggests we should have the source line first. OTOH - the compiler is talking to the user providing the source, i.e. the user's input is that source, and that's what the compiler is referring to. For `RLogger` the situation is different, and the source line is merely the context provided to the developer. So yes, I now agree 100% and never thought of this major difference. Thanks a lot for pointing that out! I suggest:

```
Info in ROOT.RDF, void ROOT::Detail::RDF::RLoopManager::Run() [tree/dataframe/src/RLoopManager.cxx:658]: Starting event loop number 0.
```
Is that any better? I'd prefer to put the message last, because the message might be multi-line, and I don't like either of these options:

1) file in new line:
```
Info in ROOT.RDF, void ROOT::Detail::RDF::RLoopManager::Run(): Starting event loop number 0.
[tree/dataframe/src/RLoopManager.cxx:658]
Info in ROOT.RDF, void ROOT::Detail::RDF::RLoopManager::Run(): And some more info.
[tree/dataframe/src/RLoopManager.cxx:660]
```
It's hard to tell where that file belong to - is that *after* the `Info in...` line? Or before?

2) file attached to last line:
```
Info in ROOT.RDF, void ROOT::Detail::RDF::RLoopManager::Run(): Starting event loop number 0.
  Some value: 17
  Other value: 42 [tree/dataframe/src/RLoopManager.cxx:658]
```
That gets rid of the ordering ambiguity, but makes it kinda weird, because it's messing with the output - is the other value `"42 [tree/dataframe/src/RLoopManager.cxx:658]"` or `42`?

**Comment by pcanal** [comment]
I would also argue that the answer might be different for different level of message.  For example, I can see a use (maybe) for the line number for Fatal and Debug message, but I don't see it for "Info" message.  I.e. some message are directed at the user (Print, Info, Warning .. maybe even Error) and other messare  are directed (indirectly) at the developers (Fatal, Debug ... maybe Error)

When reading:
```
Info in ROOT.RDF, void ROOT::Detail::RDF::RLoopManager::Run() [tree/dataframe/src/RLoopManager.cxx:658]: Starting event loop number 0.
```
I find the (relative path) filename and line number entirely redundant.  The information is already contained in the ouput message and function name (just open the file, get to the routine and search for the message).   Ambiguous messages (same text for different lines) are rare and it is even rarer that you can't figure it out.  [This applies "less" to debug message that are more likely to be terser and more repeated]

**Comment by eguiraud** [comment]
> I'd prefer to put the message last, because the message might be multi-line

That's a good argument.

Thinking about [what I'm using RLogger for in RDF](https://github.com/root-project/root/pull/6828) I tend to agree with @pcanal, users won't care about what source code line the `Info` messages come from (and if they really do, they are one grep away) but devs would hate to have to grep for a given Error/Fatal/Warning every time.

**Comment by Axel-Naumann** [comment]
> I find the (relative path) filename and line number entirely redundant. The information is already contained in the ouput message and function name (just open the file, get to the routine and search for the message).

*The* file - which file should I open?

> Ambiguous messages (same text for different lines) are rare

Define "rare".

```
io/io/src/TFileMerger.cxx:                                 Info("MergeRecursive", "could not read object for key {%s, %s}; skipping file %s",
io/io/src/TFileMerger.cxx:                              Info("MergeRecursive", "could not read object for key {%s, %s}; skipping file %s",
```
```
io/io/src/TStreamerInfo.cxx:               Info("BuildOld", "element: %s %s::%s has new type: %s/%d", element->GetTypeName(), GetName(
), element->GetName(), dm ? dm->GetFullTypeName() : TDataType::GetTypeName((EDataType)newType), newType);
io/io/src/TStreamerInfo.cxx:               Info("BuildOld", "element: %s %s::%s has new type: %s/%d", element->GetTypeName(), GetName(
), element->GetName(), dm ? dm->GetFullTypeName() : TDataType::GetTypeName((EDataType)newType), newType);
```
to name just those I immediately found in io/io :-)

I'd argue we should keep the file name:
- for debug output, be verbose because someone wanted debug output.
- for error messages, be verbose because it's an exceptional, an error case, and more information is strictly better.
So indeed, maybe for `Info` we could go with your "super-short" version.

But I dislike having the same interface behave completely differently. So what should the interface be? Maybe a new CPP macro, `R__LOG_TERSE_INFO`, that does not show the file and line and function?

**Comment by pcanal** [comment]
The one in TStreamerInfo.cxx are:
```if (gDebug>0) Info(...)```
so odd beast and/or now a Debug message.

The one in TFileMerger are likely actually Warnings rather than Info but either way, the line number is not that informative ... (the debugging that would need to happen is about the ROOT file not the source file).

Either way

> ```R__LOG_TERSE_INFO```, that does not show the file and line and function?

I suppose so .. and/or have a terse option/parameter (so that it can be used for all level)

**Comment by dpiparo** [comment]
absorbed in #6931
